main/transaction/index.ts

```diff
@@ -91,19 +91,40 @@ function populate (rawTx: TransactionData, chainConfig: Common, gas: any): Trans
 
   if (chainConfig.isActivatedEIP(1559) && gas.price.fees) {
     txData.type = intToHex(2)
+    
+    const useDappMaxFeePerGas = rawTx.maxFeePerGas && !isNaN(parseInt(rawTx.maxFeePerGas, 16))
+    const useDappMaxPriorityFeePerGas = rawTx.maxPriorityFeePerGas && !isNaN(parseInt(rawTx.maxPriorityFeePerGas, 16))
+    
+    if (useDappMaxFeePerGas || useDappMaxPriorityFeePerGas) {
+      // dapp supplied a valid value for maxFeePerGas or maxPriorityFeePerGas so we change the source flag
+      txData.gasFeesSource = GasFeesSource.Dapp
+    }
 
-    const maxPriorityFee = toBN(gas.price.fees.maxPriorityFeePerGas)
-    const maxBaseFee = toBN(gas.price.fees.maxBaseFeePerGas)
-    const maxFee = maxPriorityFee.add(maxBaseFee)
-
-    txData.maxPriorityFeePerGas = bnToHex(maxPriorityFee)
-    txData.maxFeePerGas = bnToHex(maxFee)
+    if (!useDappMaxFeePerGas) {
+      // no valid dapp-supplied value for maxFeePerGas so we calculate it from Frame or dapp-supplied values 
```

Comment:
remove the last part of this comment I think

Reply:
maxPriorityFeePerGas can still be dapp-supplied and used in the calculation, but I'll make it more clear